### PR TITLE
Update source to work with kafka-1.0.0

### PR DESCRIPTION
**Pull request by norrs**
Also fixed CI to run with all profiles.

Changes:

You now need to specify a profile when building, because of scala versions clashes and kafka versions etc. 

I didn't get a default profile to work, as toggling it to another profile, the default one was still activated sadly.

### CODE REVIEW

**Comment by HenryCaiHaiying** [comment]
@norrs I was reading http://maven.apache.org/guides/introduction/introduction-to-profiles.html to see why the default profile is always included.  Based on the wiki:
...
All profiles that are active by default are automatically deactivated when a profile in the POM is activated on the command line or through its activation config.
...
I think secor's problem is there are two activeByDefault profiles: kafka-0.10.0.1 and kafka-0.10-dev, if you fix that, would that solved the problem?

It's OK to move the activeByDefault profile to 0.11.0 or 1.0.0, but I still want to see by default the developer don't have to specify profile for the default case.  There are other tricks by using environment variables to activate profile if we have to.

**Comment by norrs** [comment]
@HenryCaiHaiying :

Default set to kafka-0.11-1.0.0 meaning scala 2_11 due to kafka version 1.0.0.

Fixed default profile and also added that travis just prints the mvn dependecy:tree in case we want to see that in build logs. 

```
$ MVN_PROFILE=release make dependency_tree | grep kafka
[INFO] +- io.confluent:kafka-avro-serializer:jar:2.0.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:2.0.1:compile
[INFO] +- org.apache.kafka:kafka_2.10:jar:0.10.2.0:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:0.10.2.0:compile

$ MVN_PROFILE=kafka-0.10.0.1 make dependency_tree | grep kafka
[INFO] +- io.confluent:kafka-avro-serializer:jar:2.0.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:2.0.1:compile
[INFO] +- org.apache.kafka:kafka_2.10:jar:0.10.0.1:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:0.10.0.1:compile

$ MVN_PROFILE=kafka-0.11-1.0.0 make dependency_tree | grep kafka
[INFO] +- io.confluent:kafka-avro-serializer:jar:2.0.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:2.0.1:compile
[INFO] +- org.apache.kafka:kafka_2.11:jar:1.0.0:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:1.0.0:compile

$ MVN_PROFILE=kafka-0.8-dev make dependency_tree | grep kafka
[INFO] +- io.confluent:kafka-avro-serializer:jar:2.0.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:2.0.1:compile
[INFO] +- org.apache.kafka:kafka_2.10:jar:0.8.2.1:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:0.8.2.1:compile


$ mvn dependency:tree  | grep kafka
[INFO] +- io.confluent:kafka-avro-serializer:jar:2.0.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:2.0.1:compile
[INFO] +- org.apache.kafka:kafka_2.11:jar:1.0.0:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:1.0.0:compile
```

**Comment by norrs** [comment]
Note I kept the release profile to remain the same kafka 0.10.2.0 with scala 2.10 !

File: README.md [hunk 1]
```diff
@@ -29,19 +29,21 @@ Edit `src/main/config/*.properties` files to specify parameters describing the e
 
 ##### Create and install jars
 ```sh
-# By default this will install the "release" (Kafka 0.10 profile)
+# By default this will install the "kafka-0.11-1.0.0" (kafka 1.0.0, scala 2.11)
```
Comment (HenryCaiHaiying) [inline README.md]: I thought you still make kafka-0.10.2 as the default?

**Comment by norrs** [comment]
@HenryCaiHaiying : Regarding CI earlier, it simply was the of the builds who had a timeout when fetching maven dependency. So basically unlucky. 

Might reconsider if we need to test all the profiles and jdk versions, or maybe have 2 travis jobs, one for testing all profiles / jdk combinations  (which seems to now take about 2 hours :O ) and one that is only running with the release profile with openjdk.

**Comment by HenryCaiHaiying** [comment]
For kafka-1.0.0 profile, do we still need to distinguish between scala_2.10 and scala_2.11?  For CI, the two travis profile options is interesting, but I don't know how much effort will it be to set it up?  I don't think we need to test different jdk versions, openjdk is fine.  So maybe three kafka profile + openjdk.  Also those profile runs are in parallel, so should more or less the same as one profile.

**Comment by norrs** [comment]
@HenryCaiHaiying : I added it in the [README.dev.md](https://github.com/pinterest/secor/pull/421/files#diff-07f96ab9a374186187d89a4caba027b1), so that should be enough. Profiles should basically just be each kafka version we want to support when I gave it some more thought.  

Also we could maybe make some shell tests in another PR who just do some smart grepping in the `mvn dependency:tree` output. 

Regarding travis setup, it looks like it only supports having one `.travis.yml` file pr repo. This is the only thing they have about speeding up builds: https://docs.travis-ci.com/user/speeding-up-the-build/#Parallelizing-your-builds-across-virtual-machines 

And we can do something like this: https://github.com/travis-ci/travis-ci/issues/3540#issuecomment-330188464 for optimizing in another pr.

**Comment by norrs** [comment]
@HenryCaiHaiying : I can update the `travis.yml` in this PR if you want to only do openjdk?

**Comment by norrs** [comment]
@HenryCaiHaiying : If you want more changes, let me know.

**Comment by HenryCaiHaiying** [comment]
Left a small comment on m2 cache.

**Comment by norrs** [comment]
@HenryCaiHaiying : I Found it from the https://docs.travis-ci.com/user/caching/#Arbitrary-directories , but it actually seems to not be doing it's job?

**Comment by norrs** [comment]
Removed the .m2 cache commit, as I didn't see the build time getting any faster ..

**Comment by HenryCaiHaiying** [comment]
This is quite a significant change, thanks for the contribution

**Pull request merged**